linux/dm-sworndisk/src/lib.rs

```diff
@@ -298,38 +378,70 @@ impl DmTargetOps for DmSwornDisk {
     type Private = DmSwornDisk;
 
     fn ctr(target: &mut Target<Self>, args: Args) -> Result<Box<Self>> {
-        // TODO: accept more arguments, e.g., root key.
-        if args.len() != 1 {
+        if args.len() != 3 {
             target.set_error(c_str!("Invalid argument count"));
             return Err(EINVAL);
         }
 
-        let Ok(raw_disk) = RawDisk::open(&args[0]) else {
+        let should_format = match args[0].to_str() {
+            Ok("create") => true,
+            Ok("open") => false,
+            _ => {
+                target.set_error(c_str!("Unsupported operation"));
+                return Err(EINVAL);
+            }
+        };
+
+        let Ok(raw_disk) = RawDisk::open(&args[1]) else {
             target.set_error(c_str!("Device lookup failed"));
             return Err(ENODEV);
         };
 
-        // TODO: use raw_disk to construct a sworndisk instance.
-        let queue = Arc::new(ReqQueue::new(raw_disk));
+        let mut root_key = AeadKey::default();
```

Comment:
Here deserves a `TODO`, giving a default root key should not be the default way. The root key must be user-given upon `dm_setup` or other approaches.